legacy/firmware/config.c

```diff
@@ -772,12 +784,69 @@ bool config_changeWipeCode(const char *pin, const char *wipe_code) {
   return sectrue == ret;
 }
 
-const uint8_t *session_getSessionId(void) {
-  if (!sessionIdCached) {
-    random_buffer(sessionId, 32);
+uint8_t session_findLRU(void) {
+  uint8_t lowest_index = MAX_SESSIONS_COUNT;
+  uint32_t lowest = sessionLRUCounter;
+  for (uint8_t i = 0; i < MAX_SESSIONS_COUNT; i++) {
+    if (sessionsCache[i].usage < lowest) {
+      lowest = sessionsCache[i].usage;
+      lowest_index = i;
+    }
+  }
+  // TODO: abort if lowest_index == MAX_SESSIONS_COUNT?
+  // TODO: it should never happen, so it is more of "internal" check
+  return lowest_index;
+}
+
+uint8_t session_findSession(const uint8_t *sessionId) {
+  for (uint8_t i = 0; i < MAX_SESSIONS_COUNT; i++) {
+    if (sessionsCache[i].usage != 0) {
```

Comment:
Where do you initialize `usage` to 0? I guess you are calling `session_clear()` somewhere during startup, but I just can't find it anywhere.

Reply:
That is a very good point. I think I did not. I have added it to config.c to a `config_init` function, which is called from the main `trezor.c`.